resources/views.py

```diff
@@ -9,3 +13,68 @@ def resources(request):
     resources = Resource.objects.all()
 
     return render(request, template, {'resources': resources})
+
+
+@login_required
+def add_resource(request):
+    """ A view allowing admin to add a resource to the resources page """
+    if not request.user.is_superuser:
+        messages.error(request, 'Sorry, only admin can add resources.')
+        return redirect(reverse('home'))
+
+    if request.method == 'POST':
+        form = ResourceForm(request.POST)
+        if form.is_valid():
+            resource = form.save()
+            messages.success(request, 'Successfully added resource!')
+            return redirect(reverse('resources'))
+        else:
+            messages.error(request, 'Failed to add resource. Please ensure the form is valid.')
+    else:
+        form = ResourceForm()
+
+    template = 'resources/add_resource.html'
```

Comment:
Not sure you need to declare a variable for this, I would just pass it directly to the `render` function.

Reply:
I've moved `ResourceForm()` to the `render` function,  is that what you mean here?